src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi

```diff
@@ -61,64 +99,132 @@ async def _handle_unary_unary_rpc(object method_handler,
                                   RPCState rpc_state,
                                   object loop):
     # Receives request message
-    cdef tuple receive_ops = (
-        ReceiveMessageOperation(_EMPTY_FLAGS),
-    )
-    await callback_start_batch(rpc_state, receive_ops, loop)
+    cdef bytes request_raw = await _receive_message(rpc_state, loop)
 
     # Deserializes the request message
-    cdef bytes request_raw = receive_ops[0].message()
-    cdef object request_message
-    if method_handler.request_deserializer:
-        request_message = method_handler.request_deserializer(request_raw)
-    else:
-        request_message = request_raw
+    cdef object request_message = deserialize(
+        method_handler.request_deserializer,
+        request_raw,
+    )
 
     # Executes application logic
-    cdef object response_message = await method_handler.unary_unary(request_message, _ServicerContextPlaceHolder())
+    cdef object response_message = await method_handler.unary_unary(
+        request_message,
+        _ServicerContext(
+            rpc_state,
+            None,
+            None,
+            loop,
+        ),
+    )
 
     # Serializes the response message
-    cdef bytes response_raw
-    if method_handler.response_serializer:
-        response_raw = method_handler.response_serializer(response_message)
-    else:
-        response_raw = response_message
+    cdef bytes response_raw = serialize(
+        method_handler.response_serializer,
+        response_message,
+    )
 
     # Sends response message
     cdef tuple send_ops = (
         SendStatusFromServerOperation(
-        tuple(), StatusCode.ok, b'', _EMPTY_FLAGS),
-        SendInitialMetadataOperation(tuple(), _EMPTY_FLAGS),
+            tuple(),
+            StatusCode.ok,
+            b'',
+            _EMPTY_FLAGS,
+        ),
+        SendInitialMetadataOperation(None, _EMPTY_FLAGS),
         SendMessageOperation(response_raw, _EMPTY_FLAGS),
     )
     await callback_start_batch(rpc_state, send_ops, loop)
 
 
+async def _handle_unary_stream_rpc(object method_handler,
+                                   RPCState rpc_state,
+                                   object loop):
+    # Receives request message
+    cdef bytes request_raw = await _receive_message(rpc_state, loop)
+
+    # Deserializes the request message
+    cdef object request_message = deserialize(
+        method_handler.request_deserializer,
+        request_raw,
+    )
+
+    cdef _ServicerContext servicer_context = _ServicerContext(
+        rpc_state,
+        method_handler.request_deserializer,
+        method_handler.response_serializer,
+        loop,
+    )
+
+    cdef object async_response_generator
+    cdef object response_message
+    if inspect.iscoroutinefunction(method_handler.unary_stream):
```

Comment:
What's the latency on this `inspect.iscoroutinefunction` call? Can this check be done ahead of time instead of in the data path?